marshmallow_recipe/bake.py

```diff
@@ -72,81 +80,81 @@ def bake_schema(
     if result := _schema_types.get(key):
         return result
 
-    fields_with_metadata = [
-        (
+    fields_type_map = get_fields_type_map(cls)
```

Comment:
Is there a reason why we pass `cls` here and not `origin`? If there is why do we use `origin` on line 95 and why we even collecting field names one more time and not using the map?

Reply:
> Is there a reason why we pass cls here and not origin?
in case with genetic type `origin` will be unsubscripted (open) generic like `Xxx[T]` without args and we can not use it to calculate actual field type and `cls` will be subscripted (closed) generic like `Xxx[int]` which has `int` as arg to replace with it all TypeVars used in fields types.

> If there is why do we use origin on line 95
subscripted generic of unsubscripted dataclass is not a dataclass. we can get fields only from origin - from unsubscripted declared type. please check PR description. There you can find explanation with examples.

> why we even collecting field names one more time and not using the map?
that map is only about fields types with access by field name. Here we collect fields to get field descriptor and field metadata to build schema